src/main/java/com/connexta/security/markings/banner/BannerCreator.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.security.markings.banner;
+
+import static org.springframework.util.StringUtils.isEmpty;
+
+import com.connexta.security.markings.api.rest.models.ISM;
+import com.connexta.security.markings.util.InputValidator;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BannerCreator {
+  private static BannerCreator instance;
+
+  public static final String DELIMITER = "//";
+  public static final String SUB_DELIMITER = "/";
+  public static final String SPACE = " ";
+  private static List<String> SPECIAL_CLASSIFICATIONS =
+      ImmutableList.of("NU", "NR", "NC", "NS", "CTS");
+
+  // singleton - make the default constructor private
+  private BannerCreator() {}
+
+  public static BannerCreator getInstance() {
+    if (instance == null) {
+      instance = new BannerCreator();
+    }
+    return instance;
+  }
+
+  public String create(ISM ism) {
+    log.trace("Entering create given ISM.");
+
+    log.trace("Input validation on ISM.");
+    InputValidator.validateIsm(ism);
+
+    StringBuilder builder = new StringBuilder();
+
+    log.trace("Append banner components.");
+    appendClassification(builder, ism);
+    appendNonUsControls(builder, ism);
+    appendFgi(builder, ism);
+    appendSap(builder, ism);
+    appendAea(builder, ism);
+    appendOtherDisseminationControls(builder, ism);
+    appendDisseminationControls(builder, ism);
+    appendSciControls(builder, ism);
+    log.debug("Banner Marking string: {}.", builder.toString());
+
+    log.trace("Exiting create given ISM.");
+    return builder.toString();
+  }
+
+  private void appendClassification(StringBuilder builder, ISM ism) {
+    String classification = ism.getClassification();
+    String ownerProducer = ism.getOwnerProducer();
+
+    boolean isJoint = Boolean.TRUE.equals(ism.isJoint());
+    boolean isFGI = !"US".equals(ownerProducer) && !"USA".equals(ownerProducer);
+
+    if (isJoint) { // is joint
```

Comment:
✏️ Nitpik: Are these comments needed at lines 67, 74, and 77 they say the same thing as the boolean name